app/services/brapi/v1/germplasm_queries.rb

```diff
@@ -122,17 +139,23 @@ def execute_statement(sql, atts)
     # at least until the previous one has been deallocated
     Thread.exclusive do
       @connection.prepare('brapi_statement', sql)
-      
-      if( (atts != nil) && !(atts.empty?) )
-        results = @connection.exec_prepared("brapi_statement", atts)
-      else
-        results = @connection.exec_prepared("brapi_statement")
+      begin
+        if( (atts != nil) && !(atts.empty?) )
+          results = @connection.exec_prepared("brapi_statement", atts)
+        else
+          results = @connection.exec_prepared("brapi_statement")
+        end
+      rescue PG::Error => e
+        @connection.exec("ROLLBACK") 
+        results = nil
+        Rails.logger.warn { "Encountered an error executing a BrAPI germplasm-related query: #{e.message} #{e.backtrace.join("\n")}" }
```

Comment:
We have error notification in place. It should be used here in addition to logging.